### ISSUE TITLE
[stable/traefik] Update KV Configs by running helm upgrade

### ISSUE THREAD

**Issue by alwinmarkcf** [helm/charts#7172]
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:

Helm:
```
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
```
K8s:
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:55:54Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-12T14:14:26Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:

stable/traefik

**What happened**:

When you change values in the values file and run:
`helm upgrade traefik` the values in etcd were not changed.

**What you expected to happen**:

If I run `helm upgrade` and afterwards:
`kubectl run  --rm -it traefik-etcd-test --image=quay.io/coreos/etcd:v3.1.8 --restart=Never --env="ETCDCTL_API=3" -l app=traefik -- etcdctl --endpoints=http://traefik-etcd-cluster-client:2379 get <traefik/dir/key> --prefix
`
I should see the new value.

**How to reproduce it** (as minimally and precisely as possible):

Install Traefik with HA enabled and ETCD as Key/Value Provider

**Anything else we need to know**:

I create a PR for this running the storeconfig job also after upgrades.

**Comment by alwinmarkcf** [comment]
I need some changes in Traefik itself or support to move on with this issue/PR :-(

**Comment by bit-herder** [comment]
can we get some movement here? We have been waiting on HA forever and even tho there are instructions in the readme, they don't work as far as i can tell. As soon as I run an upgrade my old PVC gets deleted, and then traefik wont start, so I cant run anything. Is that whats happening to everyone else?

**Comment by alwinmarkcf** [comment]
@bit-herder no the PVC is not the place where the state is saved but the key/value store. So its ok when the PVC got killed. You need a KV Provider.

This PR is just about updating the configs in the KV Store while doing an helm upgrade. But this is not working, as it would not delete keys. So I need traefik to use the config.toml and not store/import the configs to the KV Store. The only thing that should be stored in the KV Provider should be the acme certs. But those are not configs but state.

**Comment by bit-herder** [comment]
@alwinmarkcf I described my setup in another issue. https://github.com/helm/charts/issues/7774

I may just be missing something, but the KV store is up.
